src/mainwindow.cc

```diff
@@ -95,6 +96,8 @@ MainWindow::MainWindow():
     quit->setShortcut(QKeySequence::Quit);
     addAction(quit);
 
+    qApp->installEventFilter(this);
```

Comment:
Why installing the event filter on whole application? We just need the systray, not?

Reply:
Fixed this by decorating QSystemTrayIcon class (eventFilter() method necessary to make this work), not quite sure whether I should start moving more stuff into it or not (most Systray code equally codependent with MainWindow).